Jenkinsfile

```diff
@@ -37,17 +37,59 @@ spec:
       name: known-hosts
 """
 
+def kubernetes_test_config = """
+apiVersion: v1
+kind: Pod
+spec:
+  containers:
+  - name: theia
+    image: grianbrcom/theia4cobol:1.0.1
+    tty: true
+    command: [ "/bin/bash", "-c", "--" ]
+    args: [ "while true; do sleep 1000; done;" ]
+    resources:
+      limits:
+        memory: "2Gi"
+        cpu: "1"
+      requests:
+        memory: "2Gi"
+        cpu: "1"
+  - name: python
+    image: grianbrcom/python-firefox:1.0.1
+    tty: true
+    resources:
+      limits:
+        memory: "2Gi"
+        cpu: "1"
+      requests:
+        memory: "2Gi"
+        cpu: "1"
+  - name: jnlp
+    volumeMounts:
+    - name: volume-known-hosts
+      mountPath: /home/jenkins/.ssh
+  volumes:
+  - name: volume-known-hosts
+    configMap:
+      name: known-hosts
+"""
+
 def projectName = 'lsp-for-cobol'
-def targetFiles = 'lsp-core-cobol-parser/target/**'
-def kubeLabel = projectName + '_pod_' + env.BUILD_NUMBER + '_' + env.BRANCH_NAME
-kubeLabel = kubeLabel.replaceAll(/[^a-zA-Z0-9._-]+/,"")
+def kubeLabelPrefix = "${projectName}_pod_${env.BUILD_NUMBER}_${env.BRANCH_NAME}".replaceAll(/[^a-zA-Z0-9._-]+/,"")
+def kubeBuildLabel = "${kubeLabelPrefix}_build"
+def kubeTestLabel = "${kubeBuildLabel}_test"
```

Comment:
Since you are touching the kubernetes pod label, be aware that [maximum length is 63 characters](https://kubernetes.io/docs/concepts/overview/working-with-objects/labels/#syntax-and-character-set)
Constructing such long label has potential to break that limit - the pod would not be created/started

Reply:
You are right. So, this means for a three-digit build number we should have a branch name length less or equal then 35 characters. The current limit is 41. I could make a short suffix or use the same label for pods.